clang/lib/Sema/SemaSPIRV.cpp

```diff
@@ -69,6 +69,49 @@ bool SemaSPIRV::CheckSPIRVBuiltinFunctionCall(unsigned BuiltinID,
     TheCall->setType(RetTy);
     break;
   }
+  case SPIRV::BI__builtin_spirv_refract: {
+    if (SemaRef.checkArgCount(TheCall, 3))
+      return true;
+
+    ExprResult A = TheCall->getArg(0);
```

Comment:
you could do ` for (unsigned i = 0; i < TheCall->getNumArgs()-1; ++i) {` so you don't have to repeat the code for ExprResult A and B.